test/dbquery_test/test_pg_generator.py

```diff
@@ -0,0 +1,123 @@
+
+Prepare a PostgreSQL test database:
+  =# CREATE ROLE <user> LOGIN PASSWORD '<password>';
+  =# CREATE DATABASE <database> WITH OWNER <user>;
+"""
+from unittest import TestCase, skipUnless
+from os import getenv
+
+from dbquery.postgres import PostgresDB
+from dbquery.query import to_dict_formatter
+
+
+_TEST_SCHEMA = "dbquery_test"
+
+_DBQUERY_POSTGRES_TEST = "DBQUERY_POSTGRES_TEST"
+
+
+@skipUnless(
+    getenv(_DBQUERY_POSTGRES_TEST), 'PostgreSQL connection tests not enabled.')
+class PostgresTestCase(TestCase):
+    """ Adds the PostgreSQL connection to the TestCase class in the setUp
+    method and clears up the DB in tearDown.
+    """
+
+    def setUp(self):
+        """ Set up a connection and create the test schema, drop it first if
+        necessary and set it as the users default.
+
+        This of course tests if a connection can be established and if
+        Manipulation works on the connections...
+        """
+        super(PostgresTestCase, self).setUp()
+        test_dsn = getenv(_DBQUERY_POSTGRES_TEST)
+        # Set up the connection.
+        self.db = PostgresDB(test_dsn, retry=3)
+        # Drop schema if it exists and create an empty one.
+        self.db.Manipulation(
+            "DROP SCHEMA IF EXISTS {} CASCADE".format(_TEST_SCHEMA))()
+        # Create schema and set it as default.
+        self.db.Manipulation("CREATE SCHEMA {}".format(_TEST_SCHEMA))()
+        self.db.Manipulation("SET search_path TO {}".format(_TEST_SCHEMA))()
+
+    def tearDown(self):
+        super(PostgresTestCase, self).tearDown()
+        self.db.Manipulation(
+            "DROP SCHEMA IF EXISTS {} CASCADE".format(_TEST_SCHEMA))()
+        self.db.close()
+
+
+class TestInsertMany(PostgresTestCase):
+    """ Insert several items and test SelectIterator. """
+
+    def test_select_iterator(self):
+        """ Create N rows, use SelectIterator and callback function.
+        """
+        N = 10
+        test_value = "hello"
+        self.db.Manipulation("CREATE TABLE test (id INTEGER, val VARCHAR)")()
+
+        for i in range(N):
+            self.db.Manipulation(
+                "INSERT INTO test VALUES(%s, %s)")(str(i), test_value + str(i))
+
+        # SelectIterator calls this once with a generator of rows.
+        def _callback(row_generator, *args):
+            """ Handles generator which delivers individual rows. Row elements
+                are tuples.
+            """
+            row_counter = 0
+            for row in row_generator:
+                self.assertEqual(
+                    row,
+                    (row_counter, "hello" + str(row_counter)))
+                row_counter += 1
+                self.assertEqual(args[0], "hello")
+                self.assertEqual(args[1], "world")
+            self.assertEqual(row_counter, 10)
+
+        cb_args = ["hello", "world"]
+        arraysize = 2
+        select = self.db.SelectIterator(
+            "SELECT * FROM test", _callback, cb_args, arraysize)
```

Comment:
It seems the test relays on the fact that the query result is id ordered? Probably a good idea to explicitly add this then to the query...